main.py

```diff
@@ -1,38 +1,69 @@
 from transformers import AutoTokenizer, AutoModelForCausalLM
 import torch
+
 # Convert datasets into a list of strings
 def load_datasets(dataset_path):
     with open(dataset_path) as f:
         text = [l.strip() for l in f if l.strip()]
         return text 
+
 # Set up the device and model
+#Use cuda if available
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+#Using bert-base-uncased for now: https://huggingface.co/google-bert/bert-base-uncased
 tokenizer = AutoTokenizer.from_pretrained('bert-base-uncased')
-model = AutoModelForCausalLM.from_pretrained('bert-base-uncased',output_hidden_states=True ,is_decoder=True).to(device)
+model = AutoModelForCausalLM.from_pretrained('bert-base-uncased',output_hidden_states=True,is_decoder=True).to(device)
+
+#Sets up model for use instead of training
 model.eval()
-model.to(device)
+model.to(device) #Migrate model to GPU/CPU as needed
+
 # Function to tokenize and batch the text
 def tokenize_and_batch(text , batch_size: int = 16):
```

Comment:
another example where comment just essentially repeats the name of the function. recommend deleting the comment.